Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py

```diff
@@ -1955,47 +1959,72 @@ def update_remote_system_command(client, args):
     # Get the last fetch time, if exists
     last_fetch = last_run.get('time') if isinstance(last_run, dict) else None
-
+    incidents_from_previous_run = last_run.get('incidents_from_previous_run', []) if isinstance(last_run,
+                                                                                                dict) else []
     # Handle first time fetch, fetch incidents retroactively
     if last_fetch is None:
         last_fetch, _ = parse_date_range(first_fetch_time, to_timestamp=True)
 
     incidents = []
-    raw_incidents = client.get_incidents(gte_creation_time_milliseconds=last_fetch,
-                                         limit=max_fetch, sort_by_creation_time='asc')
+    if incidents_from_previous_run:
+        raw_incidents = incidents_from_previous_run
+    else:
+        raw_incidents = client.get_incidents(gte_creation_time_milliseconds=last_fetch,
+                                             limit=max_fetch, sort_by_creation_time='asc')
 
-    for raw_incident in raw_incidents:
-        incident_id = raw_incident.get('incident_id')
+    # maintain a list of non created incidents in a case of a rate limit exception
+    non_created_incidents: list = raw_incidents.copy()
+    next_run = dict()
 
-        if demisto.params().get('extra_data'):
-            incident_data = get_incident_extra_data_command(client, {"incident_id": incident_id,
-                                                                     "alerts_limit": 1000})[2].get('incident')
-        else:
-            incident_data = raw_incident
+    try:
+        for raw_incident in raw_incidents:
+            incident_id = raw_incident.get('incident_id')
 
-        sort_all_list_incident_fields(incident_data)
+            if demisto.params().get('extra_data'):
+                incident_data = get_incident_extra_data_command(client, {"incident_id": incident_id,
+                                                                         "alerts_limit": 1000})[2].get('incident')
+            else:
+                incident_data = raw_incident
 
-        incident_data['mirror_direction'] = MIRROR_DIRECTION.get(demisto.params().get('mirror_direction', 'None'), None)
-        incident_data['mirror_instance'] = demisto.integrationInstance()
+            sort_all_list_incident_fields(incident_data)
 
-        description = raw_incident.get('description')
-        occurred = timestamp_to_datestring(raw_incident['creation_time'], TIME_FORMAT + 'Z')
-        incident = {
-            'name': f'#{incident_id} - {description}',
-            'occurred': occurred,
-            'rawJSON': json.dumps(incident_data),
-        }
+            incident_data['mirror_direction'] = MIRROR_DIRECTION.get(demisto.params().get('mirror_direction', 'None'),
+                                                                     None)
+            incident_data['mirror_instance'] = demisto.integrationInstance()
```

Comment:
same, pass the `demisto.integrationInstance()` to `fetch_incidents` as `integration_instance`

Reply:
Done